src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValue.cs

```diff
@@ -38,13 +40,46 @@ public abstract partial class JsonValue : JsonNode
                     return null;
                 }
 
-                if (element.ValueKind == JsonValueKind.Object || element.ValueKind == JsonValueKind.Array)
+                VerifyJsonElementIsNotArrayOrObject(ref element);
```

Comment:
Should we be creating a `JsonValueTrimmable` instance in the case where it is a JsonElement instead?